src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -441,6 +441,7 @@ def load_arguments(self, _):
                    min_api="2019-10-01")
         c.argument('what_if', arg_type=deployment_what_if_type)
         c.argument('proceed_if_no_change', arg_type=deployment_what_if_proceed_if_no_change_type)
+        c.argument('mode', arg_type=get_enum_type(DeploymentMode, default='incremental'), help='The mode that is used to deploy resources. This value can be either Incremental or Complete. In Incremental mode, resources are deployed without deleting existing resources that are not included in the template. In Complete mode, resources are deployed and existing resources in the resource group that are not included in the template are deleted. Be careful when using Complete mode as you may unintentionally delete resources.')
```

Comment:
> default='incremental'

Why is the default value of `DeploymentMode` is `incremental`?

Reply:
If the `mode` parameter is not passed, the service return mode cannot be `None`, and the original logic is incremental by default.
https://github.com/Azure/azure-cli/blob/4922433eef479449959a5817d237d2326ec30c40/src/azure-cli/azure/cli/command_modules/resource/custom.py#L741-L744